### ISSUE TITLE
Traceback error

### ISSUE THREAD

**Issue by ccfiel** [JackDinn/tbtray#3]
Hello @JackDinn,

I got this error

```
Traceback (most recent call last):
  File "./tbtray.py", line 586, in <module>
    main()
  File "./tbtray.py", line 580, in main
    ui = MainApp()
  File "./tbtray.py", line 305, in __init__
    checksettings()
  File "./tbtray.py", line 111, in checksettings
    with open(my_file, 'w') as configfile:
TypeError: invalid file: PosixPath('/home/chris/.config/tbtray/settings.ini')
```
any ideas?

**Comment by JackDinn** [comment]
does `/home/chris/.config/tbtray/settings.ini` exist ?  it  should have created the .config/tbtray folder and then copied `tbtray/settings.ini` into it. 
```
98 my_file = Path(str(Path.home()) + '/.config/tbtray/settings.ini')
99 if not my_file.is_file(): copyfile('settings.ini', str(my_file))
```

You can try manually creating and copy across the settings.ini file but im just not sure why it did not create it itself.

**Comment by JackDinn** [comment]
it should also have created the folder `~/.config/tbtray/icons` so if that does not exist then you better make it yourself as well.

**Comment by alok1111** [comment]
```
@user@asus:~/tmp/tbtray$ ./tbtray/tbtray.py 
Traceback (most recent call last):
  File "./tbtray/tbtray.py", line 609, in <module>
    main()
  File "./tbtray/tbtray.py", line 603, in main
    ui = MainApp()
  File "./tbtray/tbtray.py", line 279, in __init__
    checksettings()
  File "./tbtray/tbtray.py", line 89, in checksettings
    with open(my_file, 'w') as configfile:
TypeError: invalid file: PosixPath('/home/user/.config/tbtray/settings.ini')
user@asus:~/tmp/tbtray$ nano /home/user/.config/tbtray/settings.ini
```
settings.ini exists and nano command opens it.
What can I do with that?

**Comment by JackDinn** [comment]
hmm odd one. what are the permissions on the settings.ini file?

Maybe try deleting it and then replacing it with a copy from https://github.com/JackDinn/tbtray/blob/master/settings.ini

**Comment by alok1111** [comment]
```
user@asus:~/tmp/tbtray$ ls -l /home/user/.config/tbtray/settings.ini
-rw-r--r-- 1 user user 260 янв  3 22:44 /home/user/.config/tbtray/settings.ini
```
Tried to delete and copy again(even downloaded Raw from the link you provided). Unfortunately this didn't help.

**Comment by JackDinn** [comment]
needs to be writable, try , 

chmod 777 /home/user/.config/tbtray/settings.ini

**Comment by alok1111** [comment]
```
user@asus:~/tmp$ ls -l /home/user/.config/tbtray/settings.ini 
-rwxrwxrwx 1 user user 259 янв  3 23:26 /home/user/.config/tbtray/settings.ini
user@asus:~/tmp$ python3.5 -V
Python 3.5.3
```
Still crashing.

**Comment by JackDinn** [comment]
so what do you get when you run a python3 terminal and do
```
f = open('/home/user/.config/tbtray/settings.ini','r')
f.read()
f.close()
```

**Comment by alok1111** [comment]
```
user@asus:~/tmp$ python3
Python 3.5.3 (default, Sep 27 2018, 17:25:39) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> f = open('/home/user/.config/tbtray/settings.ini','r')
>>> f.read()
'[popup]\nopacity = 100\non = 1\nsoundpath = res/popup.wav\nsoundon = 1\nx = 1875\nfavicons = 1\nduration = 10\nfixedwidth = 1\ntop = 1\n\n[ticks]\nminimizetotray = 1\nshowcount = 1\n\n[icons]\ndefault = res/thunderbird.png\nnotify = res/mail.png\ncolour = #ffffff\n\n[profiles]\n\n'
>>> f.close()
```

**Comment by JackDinn** [comment]
i wonder if its because im using python 3.7.1

does this throw an error in a python terminal ?
`with open('/home/user/.config/tbtray/settings.ini', 'r') as x:`

**Comment by JackDinn** [comment]
make a file containing this and save it as pytest.py then run it in the bash terminal with 
>python ~/pytest.py
```
#!/usr/bin/env python3
with open('/home/greg/.config/tbtray/settings.ini', 'r') as x:
    z = x.read()
print (z)
```
it should give the ini file as output.

**Comment by JackDinn** [comment]
ok so doing a search for this error it does indeed look like you need at least python 3.6 

> This issue is caused by incompatibility of pathlib and python3.5.
> This repo requires python3.6. Sorry for not indicating it in README.

I had no idea about this, i shall add it to the README if you can confirm that a python update fixes this problem.

**Comment by alok1111** [comment]
Hi, i've built Python 3.7.2 from sources and it worked!
Also PyQT5 was required to install.
BTW, I noticed that:
1. Very first start crashes with log.txt opening error, but it creates log.txt in tbtray.py root folder and there is no such error next time.
2. First start without errors shows notification about settings and then tbtray stops. I've tried to start thunderbird and then tbtray.py and tbtray worked fine.
3. Please rename General tab in Settings. It is called 'Genral' for now)

**Comment by JackDinn** [comment]
Thx for getting back to me. I did test on a VM of several distros and DE's and they all had pyqt5 (well the parts needed to run tbtray) after a default installation.
Which distro & DE are you using?

I have not seen this log.txt creation error or seen your second error on any of the VM tests iv done but i shall take another look.

O spelling , opps, I do try but dyslexia sometimes still wins ^^. Will be corrected next update.

Cheers.

**Comment by alok1111** [comment]
Debian GNU/Linux 9.6 (stretch) stable with KDE

**Comment by JackDinn** [comment]
I have always avoided straight Debian as installing everything spins my head. However im sure there will be distros that don't have any pyqt so i shall add it to the requirements in the readme.

Im just testing installing on a clean VM of manjaro gnome & manjaro kde but i have not been able to reproduce these errors.

I would really need more info, like a terminal error output. However i have stuck a try:/except: around the logger just in case it helps after the next update, although i could remove the logger as it was just for me anyhow.

To try to reproduce the other error i have been deleting (renaming) the ~/.config/tbtray folder and then run tbtray to see if it shows the account settings warning and does crash but again it works for me and i can not reproduce a crash/close.

**Comment by JackDinn** [comment]
Actually not sure why i opened this thread again since the actual OP's problem is now closed (python ver 3.6+ is required).

I have just done an update but i have not addressed your last bug.

I shall close this post, but if you do reproduce either of these other errors with a terminal error output please do start a new post.

Thx for your input :+1:
